lab_1/update-pipeline-definition.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash -
+#===============================================================================
+#
+#          FILE: update-pipeline-definition.sh
+#
+#         USAGE: ./update-pipeline-definition.sh
+#
+#   DESCRIPTION: 
+#
+#       OPTIONS: ---
+#  REQUIREMENTS: ---
+#          BUGS: ---
+#         NOTES: ---
+#        AUTHOR: YOUR NAME (), 
+#  ORGANIZATION: 
+#       CREATED: 03/15/2023 10:12:08 PM
+#      REVISION:  ---
+#===============================================================================
+
+set -o nounset                                  # Treat unset variables as an error
+
+checkJQ() {
+  # jq test
+  type jq >/dev/null 2>&1
+  exitCode=$?
+	jqDependency=1
+	
+  if [ "$exitCode" -ne 0 ]; then
+    printf "  ${red}'jq' not found! (json parser)\n${end}"
+    printf "    Ubuntu Installation: sudo apt install jq\n"
+    printf "    Redhat Installation: sudo yum install jq\n"
+    jqDependency=0
+  fi
+
+  if [ "$jqDependency" -eq 0 ]; then
+    printf "${red}Missing 'jq' dependency, exiting.\n${end}"
+    exit 1
+  fi
+}
+
+checkJQ
+
+pipelineJson=$1
+branch=''
+owner=''
+configuration=''
+pollForSourceChanges=''
+
+for (( i=2; i<=$#; i++  ))
+do
+	j=$((i+1))
+	case ${!i} in
+		--branch)
+			branch=${!j}
+			;;
+		--owner)
+			owner=${!j}
+			;;
+		--configuration)
+			configuration=${!j}
+			;;
+		--poll-for-source-changes)
+			pollForSourceChanges=${!j}
+	esac
+done
+
+jq 'del(.metadata)' "$pipelineJson" > tmp.??.json && mv tmp.??.json "$pipelineJson"
+
+if [[ $( jq '.pipeline.version' $pipelineJson )  == null ]]
+then
+	echo 'version is missing in json file'
+	exit 1
+else
+nextVersion=$( jq '.pipeline.version+1' $pipelineJson )
+jq --arg nextVersion $nextVersion ".* { pipeline: { version: $nextVersion } }" "$pipelineJson" > tmp.??.json && mv tmp.??.json "$pipelineJson"
+fi
```

Comment:
`jq '.pipeline.version += 1'` - this will replace old version with new one

Reply:
fixed, thanks